gdb/riscv-tdep.c

```diff
@@ -1362,19 +1266,155 @@ riscv_gdbarch_init (struct gdbarch_info info,
       for (auto reg_info = riscv_reg_info.begin();
            reg_info != riscv_reg_info.end(); ++reg_info)
         {
-          riscv_reg_map[reg_info->number] = &(*reg_info);
           for (auto name = reg_info->names.begin() + 1;
                name != reg_info->names.end(); ++name)
             user_reg_add (gdbarch, *name,
                           value_of_riscv_user_reg, &reg_info->number);
         }
+    }
+
+  return use_tdesc_registers;
+}
 
-      set_gdbarch_num_regs (gdbarch, RISCV_NUM_REGS);
-      set_gdbarch_sp_regnum (gdbarch, RISCV_SP_REGNUM);
-      set_gdbarch_pc_regnum (gdbarch, RISCV_PC_REGNUM);
-      set_gdbarch_ps_regnum (gdbarch, RISCV_FP_REGNUM);
-      set_gdbarch_deprecated_fp_regnum (gdbarch, RISCV_FP_REGNUM);
+static struct gdbarch *
+riscv_gdbarch_init (struct gdbarch_info info,
+		    struct gdbarch_list *arches)
+{
+  /*
+   * Note that this function gets called multiple times, for different reasons.
+   * It gets called when we connect to a debug server, and called twice when
+   * the `file` command is used. In the very last case we don't get a
+   * target_desc, even if one was provided when connecting to the target.
+   */
+
+  struct gdbarch_tdep *tdep;
+  const struct bfd_arch_info *binfo = info.bfd_arch_info;
+
+  int abi;
+
+  /* For now, base the abi on the elf class.  */
+  /* Allow the ELF class to override the register size. Ideally the target
+   * (OpenOCD/spike/...) would communicate the register size to gdb instead. */
+  abi = RISCV_ABI_FLAG_RV32I;
```

Comment:
This line appears unnecessary, as either abi is set below, or we call internal_error.

Reply:
Removed. Thanks for reviewing!